internal/database/postgres.go

```diff
@@ -309,66 +311,125 @@ FROM TempDownsampled;
 }
 
 // TODO: consider returning workstationGroup
-func (conn PostgresConn) LatestData() ([]uplink.GpuStatsUpload, error) {
-	// we pull Uuid twice so we can put one into Stat sample and the other into Info
-	rows, err := conn.db.Query(`SELECT g.Machine, g.Uuid, g.Uuid, g.Name,
-			g.Brand, g.DriverVersion, g.MemoryTotal,
-			s.MemoryUtilisation, s.GpuUtilisation, s.MemoryUsed,
-			s.FanSpeed, s.Temp, s.MemoryTemp, s.GraphicsVoltage,
-			s.PowerDraw, s.GraphicsClock, s.MaxGraphicsClock,
-			s.MemoryClock, s.MaxMemoryClock
+func (conn PostgresConn) LatestData() (broadcast.Workstations, error) {
+	// pull all the machines in, then gpus for those machines
+	// has to all be done in a transaction to avoid race conditions
+	tx, err := conn.db.Begin()
+	if err != nil {
+		return nil, err
+	}
+
+	groups := make(map[string][]broadcast.Workstation)
+	machines, err := tx.Query(`SELECT GroupName, Hostname, CPU, Motherboard,
+		Notes, LastSeen
+		FROM Machines`)
+	if err != nil {
+		return nil, errors.Join(err, tx.Rollback())
+	}
+
+	for machines.Next() {
+		var groupName *string
```

Comment:
Use default values in the SQL query instead rather than custom logic.

Reply:
Done in f369989